Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -304,6 +305,146 @@ public OrderTicket StopLimitOrder(Symbol symbol, int quantity, decimal stopPrice
+            var option = (Option)Securities[optionSymbol];
+
+            var request = CreateSubmitOrderRequest(OrderType.OptionExercise, option, quantity, tag);
+
+            //Initialize the exercise order parameters
+            var preOrderCheckResponse = PreOrderChecks(request);
+            if (preOrderCheckResponse.IsError)
+            {
+                return OrderTicket.InvalidSubmitRequest(Transactions, request, preOrderCheckResponse);
+            }
+
+            //Add the order and create a new order Id.
+            var ticket = Transactions.AddOrder(request);
+
+            // Wait for the order event to process, only if the exchange is open
+            if (!asynchronous)
+            {
+                Transactions.WaitForOrder(ticket.OrderId);
+            }
+
+            return ticket;
+        }
+
+        // Support for option strategies trading
+
+        /// <summary>
+        /// Buy Option Strategy (Alias of Order)
+        /// </summary>
+        /// <param name="strategy">Specification of the strategy to trade</param>
+        /// <param name="quantity">Quantity of the strategy to trade</param>
+        /// <returns>Sequence of order ids</returns>
+        public IEnumerable<OrderTicket> Buy(OptionStrategy strategy, int quantity)
+        {
+            return Order(strategy, Math.Abs(quantity));
+        }
+
+        /// <summary>
+        /// Sell Option Strategy (alias of Order)
+        /// </summary>
+        /// <param name="strategy">Specification of the strategy to trade</param>
+        /// <param name="quantity">Quantity of the strategy to trade</param>
+        /// <returns>Sequence of order ids</returns>
+        public IEnumerable<OrderTicket> Sell(OptionStrategy strategy, int quantity)
+        {
+            return Order(strategy, Math.Abs(quantity) * -1);
+        }
+
+        /// <summary>
+        ///  Issue an order/trade for buying/selling an option strategy 
+        /// </summary>
+        /// <param name="strategy">Specification of the strategy to trade</param>
+        /// <param name="quantity">Quantity of the strategy to trade</param>
+        /// <returns>Sequence of order ids</returns>
+        public IEnumerable<OrderTicket> Order(OptionStrategy strategy, int quantity)
+        {
+            return GenerateOrders(strategy, quantity);
+        }
+
+        private IEnumerable<OrderTicket> GenerateOrders(OptionStrategy strategy, int strategyQuantity)
+        {
+            var orders = new List<OrderTicket>();
+
+            // setting up the tag text for all orders of one strategy
+            var strategyTag = strategy.Name + " (" + strategyQuantity.ToString() + ")";
+
+            // walking through all option legs and issuing orders
+            if (strategy.OptionLegs != null)
+            {
+                foreach (var optionLeg in strategy.OptionLegs)
+                {
+                    var optionSeq = Securities.Where(kv => kv.Key.Underlying == strategy.Underlying &&
```

Comment:
Have to think about it. The idea behind OptionStrategy is to introduce the concept itself to the enduser. So that the user could just create e.g. Condor w/o thinking too much if it is a valid condor - we check it. Plus, it simplifies understanding of the strategy itself, because user can buy a condor or sell two condors. In the future enduser will see that strategy itself can have bid and ask, can have greeks, and we will route the entire strategy to the broker taking an opportunity for broker's better execution capabilities of strats.